ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -456,11 +456,31 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
       final Pair<String, Schema> value = process(node.getValue(), context);
       switch (node.getSign()) {
         case MINUS:
-          // this is to avoid turning a sequence of "-" into a comment (i.e., "-- comment")
-          final String separator = value.getLeft().startsWith("-") ? " " : "";
-          return new Pair<>("-" + separator + value.getLeft(), value.getRight());
+          if (DecimalUtil.isDecimal(value.getRight())) {
+            return new Pair<>(
+                String.format(
+                    "(%s.negate(new MathContext(%d, RoundingMode.UNNECESSARY)))",
+                    value.getLeft(),
+                    DecimalUtil.precision(value.getRight())),
```

Comment:
nit: unrelated to this PR, but reading the implementation for `DecimalUtil.precision()` is confusing since the intermediate variable is called `scaleString`.

Reply:
oops! I'll fix that